### PR TITLE
ROC-4403 full admission by set date

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-4403

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/integration-test/tests/citizen/defence/steps/defence.ts [hunk 1]
```diff
@@ -143,9 +148,12 @@ export class DefenceSteps {
   }
 
   addTimeLineOfEvents (timeline: Timeline): void {
+    I.wait(5)
```
Comment (markry-kainos) [inline src/integration-test/tests/citizen/defence/steps/defence.ts]: ffs...will remove these! :-D

File: src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    } else {
+      acceptPaymentMethodPage.chooseNo()
+      taskListPage.selectProposeAnAlternativeRepaymentPlan()
+      switch (paymentOption) {
+        case PaymentOption.IMMEDIATELY:
+          paymentOptionPage.chooseImmediately()
+          break
+        case PaymentOption.BY_SET_DATE:
+          paymentOptionPage.chooseFullBySetDate()
+          paymentDatePage.enterDate('2025-01-01')
```
Comment (pragnesh-patel) [inline src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts]: add hardcoded date into testdata

File: src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  viewClaimFromDashboard (claimRef: string, shouldRespondToOffer: boolean): void {
+    I.click('My account')
+    I.see('Your money claims account')
+    I.click(claimRef)
+    if (shouldRespondToOffer) {
+      I.click('View and respond to the offer')
+    }
+  }
+
+  acceptSettlement (shouldAcceptPaymentMethod: boolean, paymentOption: PaymentOption): void {
```
Comment (pragnesh-patel) [inline src/integration-test/tests/citizen/claimantResponse/steps/claimant-reponse.ts]: maybe split this function into two, as function name suggests accepting settlement when one flow does accepSettlementAndPaymentMethod, rejectSettlementSuggestPaymentPlan

File: src/integration-test/tests/citizen/claimantResponse/steps/defendant.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    defenceSteps.loginAsDefendant(defendantEmail)
+    I.click(claimRef)
+    I.click('Respond to claim')
+    I.dontSee('COMPLETE')
+    defenceSteps.confirmYourDetails(createDefendant(defendantType))
+    defenceSteps.requestNoExtraTimeToRespond()
+    defenceSteps.rejectAllOfClaimAsDisputeClaim()
+    defendantSteps.selectTaskWhyDoYouDisagreeWithTheClaim()
+    yourDefencePage.enterYourDefence('Defendant rejects all the claim because...')
+    timelineEventsPage.enterTimelineEvent(0, '1/1/2000', 'something')
```
Comment (pragnesh-patel) [inline src/integration-test/tests/citizen/claimantResponse/steps/defendant.ts]: use test data for Timeline

File: src/integration-test/tests/citizen/claimantResponse/steps/defendant.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    timelineEventsPage.submitForm()
+    evidencePage.enterEvidenceRow('CONTRACTS_AND_AGREEMENTS', 'correspondence', 'have this evidence')
```
Comment (pragnesh-patel) [inline src/integration-test/tests/citizen/claimantResponse/steps/defendant.ts]: Use Test data for evidence

File: src/integration-test/tests/citizen/claimantResponse/pages/claimant-check-and-send.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  signerRole: 'input[id=signerRole]'
+}
+
+const buttons = {
+  submit: 'input[type=submit]'
+}
+
+export class ClaimantCheckAndSendPage {
+
+  signStatementOfTruthAndSubmit (signerName: string, signerRole: string): void {
```
[the remark this answers is not in the document]
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/pages/claimant-check-and-send.ts]: forgot about this! :-D will fix

**Comment by markry-kainos** [comment]
@pragnesh-patel i'm getting a green build post review fixes! can you review again pls 👍 :)

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+  testData.claimantPaymentOption = PaymentOption.BY_SET_DATE
+  const claimantResponseTestData = new ClaimantResponseTestData()
+  // as defendant
+  helperSteps.finishResponseWithFullAdmission(testData)
+  I.click('Sign out')
+  // as claimant
+  userSteps.login(testData.claimantEmail)
+  claimantResponseSteps.acceptSettlementFromDashboardWhenRejectPaymentMethod(testData, claimantResponseTestData)
+  I.see('Error: (/usr/src/app/src/main/features/claimant-response/views/check-and-send.njk)')
+  // checkAndSendPage.verifyFactsForSettlement()
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: Please remove if not needed any more

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  // as defendant
+  helperSteps.finishResponseWithFullAdmission(testData)
+  I.click('Sign out')
+  // as claimant
+  userSteps.login(testData.claimantEmail)
+  claimantResponseSteps.acceptSettlementFromDashboardWhenRejectPaymentMethod(testData, claimantResponseTestData)
+  checkAndSendPage.verifyFactsForSettlement()
+  checkAndSendPage.checkFactsTrueAndSubmit()
+  I.see('422 - "courtDetermination.courtPaymentIntention : must not be null"')
+  // confirmationPage.clickGoToYourAccount()
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: please remove if not needed any more

File: src/integration-test/tests/citizen/claimantResponse/data/ClaimantResponseTestData.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
+
+interface PageSpecificValues {
+  paymentDatePage_enterDate: string
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/claimantResponse/data/ClaimantResponseTestData.ts]: I'm not sure I like the _ to break the variable name. Can we keep it camelCase for consistency? if we're going to change the convention for variables probably best to discuss first at a refinement or tech demo session

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+const helperSteps: Helper = new Helper()
+const userSteps: UserSteps = new UserSteps()
+const claimantResponseSteps: ClaimantResponseSteps = new ClaimantResponseSteps()
+const checkAndSendPage: ClaimantCheckAndSendPage = new ClaimantCheckAndSendPage()
+const confirmationPage: ClaimantConfirmation = new ClaimantConfirmation()
+
+Feature('Claimant Response').retry(3)
+
+Scenario('I can as a claimant view the defendants full admission with immediate payment @citizen @admissions', async (I: I) => {
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: should this have the citizen tag? would it not be better to just have admissions one and update the command that runs the tests?

File: src/integration-test/tests/citizen/claimantResponse/reject_test.ts [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+    I.click(testData.claimRef)
+    I.see(testData.claimRef)
+    I.see('Claim status')
+    I.see('The defendant has rejected your claim')
+    I.see(`They said they dispute your claim.`)
+    I.click('Sign out')
+  })
+
+Scenario(
+  'As a claimant I can reject the claim as I have paid less than the amount claimed @citizen @admissions @error',
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/claimantResponse/reject_test.ts]: what is the @error do? is it ever used?

File: src/integration-test/tests/citizen/defence/data/DefenceTest.ts [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+import { PartyType } from 'integration-test/data/party-type'
+import { createClaimData } from 'integration-test/data/test-data'
+import { PaymentOption } from 'integration-test/data/payment-option'
+import { DefenceType } from 'integration-test/data/defence-type'
+import { Helper } from 'integration-test/tests/citizen/endToEnd/steps/helper'
+
+const helperSteps: Helper = new Helper()
+
+interface PageSpecificValues {
+  paymentDatePage_enterDate: string
```
Comment (robertparkinson) [inline src/integration-test/tests/citizen/defence/data/DefenceTest.ts]: as above

File: src/integration-test/tests/citizen/claimantResponse/reject_test.ts (hunk 11 above)
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/reject_test.ts]: Used @error where assertions are made on error state - so these should break when code fixed and the error tag removed.

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts (hunk 10 above)
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: if we tag them all citizen - we can at least run them all with one tag. then adding more tags to filter from that. i did more 'refactoring' than i thought i was going to do as i thought we would have session to discuss how we wanted to architect the e2e. we should talk about tagging too.

File: src/integration-test/tests/citizen/claimantResponse/data/ClaimantResponseTestData.ts (hunk 9 above)
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/data/ClaimantResponseTestData.ts]: Will change :)

File: src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts (hunk 8 above)
Comment (markry-kainos) [reply src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: hopefully when fixed these can just be commented out for test to pass - i can get rid of though? as will need looking at anyway

Comment (robertparkinson) [reply src/integration-test/tests/citizen/claimantResponse/full_admission_test.ts]: ah ok. this is one of the flows that error! understood

**Pull request merged**